make distclean error

### Describe the bug

In linux,  `make distclean`  got error:

```
make[1]: Entering directory '/home/pi/build/pjproject/pjsip-apps/src/swig'
make  --no-print-directory -C python distclean
rm -rf _pjsua2.so pjsua2_wrap.cpp pjsua2_wrap.h pjsua2.py build *.pyc
rm -f gcc.exe g++.exe
make  --no-print-directory -C java distclean
Makefile:115: *** Cannot determine JDK include/library path. Please define JAVA_HOME envvar.  Stop.
make[1]: *** [Makefile:27: java] Error 2
make[1]: Leaving directory '/home/build/pjproject/pjsip-apps/src/swig'
```

### Steps to reproduce

1.   `./configure`
2.  `make distclean`

### PJSIP version

2.15-dev (latest master branch)

### Context

-

### Log, call stack, etc

```shell
-
```

--- a/pjsip-apps/src/swig/java/Makefile
+++ b/pjsip-apps/src/swig/java/Makefile
@@ -40,7 +40,13 @@ else
 endif
 
 # Get JDK location
-ifeq ("$(JAVA_HOME)","")
+CHECK_JDK := 1
+ifeq ($(findstring clean,$(MAKECMDGOALS)),clean)
+  CHECK_JDK := 0
+endif
+
+ifeq ($(CHECK_JDK), 1)
+ ifeq ("$(JAVA_HOME)","")
 
   verify_jdk  = $(shell test -d $(1)/include && test -d $(1)/lib && \
                 	echo inclib1) \
@@ -121,7 +127,7 @@ ifeq ("$(JAVA_HOME)","")
     $(warning Cannot determine JDK binary path. Set to $(JAVA_BIN))
   endif
 
-else
+ else
 
   JAVA_INC := $(JAVA_HOME)/include
   JAVA_LIB := $(JAVA_HOME)/lib
@@ -131,6 +137,7 @@ else
     JAVA_BIN := $(JAVA_HOME)
   endif
 
+ endif
 endif
 
 # Env settings, e.g: path to SWIG, JDK, java(.exe), javac(.exe)

